daemon/access.go

```diff
@@ -244,3 +245,31 @@ func (ac themesAuthenticatedAccess) CheckAccess(d *Daemon, r *http.Request, ucre
 
 	return Unauthorized("access denied")
 }
+
+func requirePromptingApiAccessImpl(d *Daemon, ucred *ucrednet) *apiError {
+	if ucred == nil {
+		return Forbidden("access denied")
+	}
+
+	switch ucred.Socket {
+	case dirs.SnapdSocket:
+		// Allow access on main snapd.socket
```

Comment:
Does snapd-desktop-integration always access these endpoints via the `snapd-snap.socket`? If so, should we deny access on `snapd.socket`?

Reply:
Yes it does, and it has no access to `snapd.socket` (this requires the snapd-control interface). AIUI the `snapd-snap.socket` is the socket that confined apps use to access snapd and it has a subset of what is available on the main socket.